src/credentials/FabricTable.cpp

```diff
@@ -68,10 +68,23 @@ CHIP_ERROR FabricInfo::CommitToStorage(PersistentStorageDelegate * storage)
     ReturnErrorOnFailure(
         storage->SyncSetKeyValue(keyAlloc.FabricRCAC(mFabric), mRootCert.data(), static_cast<uint16_t>(mRootCert.size())));
 
-    // If we stop storing ICA certs when empty, update LoadFromStorage
-    // accordingly to check for CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND.
-    ReturnErrorOnFailure(
-        storage->SyncSetKeyValue(keyAlloc.FabricICAC(mFabric), mICACert.data(), static_cast<uint16_t>(mICACert.size())));
+    // Workaround for the fact that some storage backends do not allow storing
+    // a nullptr with 0 length.  See
+    // https://github.com/project-chip/connectedhomeip/issues/16030.
+    if (!mICACert.empty())
+    {
+        ReturnErrorOnFailure(
+            storage->SyncSetKeyValue(keyAlloc.FabricICAC(mFabric), mICACert.data(), static_cast<uint16_t>(mICACert.size())));
+    }
+    else
+    {
+        // Make sure there is no stale data.
+        CHIP_ERROR err = storage->SyncDeleteKeyValue(keyAlloc.FabricICAC(mFabric));
```

Comment:
Why do we need this? If no one put the key in there, there shouldn't be any "stale" data?

Reply:
There might be an old key, with a nonempty ICAC, for the same fabric, from a previous NOC (and now we are storing after UpdateNOC).